consul/src/main/java/com/linecorp/armeria/common/consul/Check.java

```diff
@@ -0,0 +1,493 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.consul;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import org.checkerframework.checker.index.qual.Positive;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * TODO Change to builder style.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Check {
+
+    @Nullable
+    @JsonProperty("Node")
+    private String node;
+
+    @Nullable
+    @JsonProperty("CheckID")
+    private String checkId;
+
+    @Nullable
+    @JsonProperty("Name")
+    private String name;
+
+    @Nullable
+    @JsonProperty("ID")
+    private String id;
+
+    @Nullable
+    @JsonProperty("Interval")
+    private String interval = "10s";
+
+    @Nullable
+    @JsonProperty("Notes")
+    private String notes;
+
+    @Nullable
+    @JsonProperty("DeregisterCriticalServiceAfter")
+    private String deregisterCriticalServiceAfter;
+
+    @Nullable
+    @JsonProperty("Args")
+    private String[] args;
+
+    @Nullable
+    @JsonProperty("AliasNode")
+    private String[] aliasNode;
+
+    @Nullable
+    @JsonProperty("DockerContainerID")
+    private String dockerContainerID;
```

Comment:
Seems strange to have this on checks that aren't Docker containers

Similarly for non overlapping tcp, grpc, http, and shell properties

Reply:
I just meant that, in practice, these check fields would never overlap. You cannot easily have an Consul service that is both a Docker container and not meanwhile running a gRPC service and also not, etc.

Reply:
Yes, I know about that. However, consul supports the API without constraint. Maybe consul has the logic. We might control them. Fortunately, we suppose to support only one check at this time that is the HTTP.
If you have any good ideas, please suggest them to us freely.

Reply:
Okay, HTTP makes sense. 

What about `DockerContainerID`? Armeria need not be running in a Docker container

---

I guess my point is that this seem to be an entirely external library that you're building in here just for purposes of service discovery